numpy/fft/_pocketfft.py

```diff
@@ -67,11 +67,19 @@ def _raw_fft(a, n, axis, is_real, is_forward, inv_norm):
             a = z
 
     if axis == a.ndim-1:
-        r = pfi.execute(a, is_real, is_forward, fct)
+        r = pfi.execute(a, is_real, is_forward, fct, out)
     else:
         a = swapaxes(a, axis, -1)
-        r = pfi.execute(a, is_real, is_forward, fct)
+        if out is not None:
+            out.resize(a.shape)
+        r = pfi.execute(a, is_real, is_forward, fct, out)
         r = swapaxes(r, axis, -1)
+        if out is not None:
+            out.resize(r.shape)
+            # This extra copy is unfortunately needed if we want `out`
+            # to retain its original shape while having the correct values.
+            copyto(out, r)
+            r = out
```

Comment:
This is contrary to the regular behaviour of `out` - you *must* ensure `out` stays the same object with the same memory layout. Note that in principle, that will be automatic -- if you swapped the axis above, then the swapped case is a view of the original `out`, so data will just be written in there. I.e., it should be possible to remove this stanza (as in the suggestion I gave above).